### ISSUE TITLE
assetLoader android path issue when bundling

### ISSUE THREAD

**Issue by grahamlyus** [callstack/haul#367]
### Current Behavior
Importing images from the root folder or from a relative path outside of the root folder does not work when bundling for android. This works in a project created with `react-native init`

I've created a repo (https://github.com/grahamlyus/ImageHaul) that reproduces the issue (make sure the haul server is not running, otherwise assets load from localhost:8081).

Running `yarn haul bundle --platform android --dev true` from the `mobile` directory produces:
```
dist
├── drawable-mdpi
│   ├── _common_relative.jpg
│   ├── _root.jpg
│   └── images_child.jpg
└── index.android.bundle

1 directory, 4 files
```
Note that the root and relative images have an underscore prepended.

The output should be:
```
dist
├── drawable-mdpi
│   ├── common_relative.jpg
│   ├── images_child.jpg
│   └── root.jpg
└── index.android.bundle

1 directory, 4 files
```

### Expected Behavior
Importing images from anywhere configured by webpack should work in all modes.

Replacing the construction of `longname` as below solves the problem by removing the leading `../` and not adding a leading underscore if the asset is in the root folder:
```
  const normalizedUrl = url.replace(new RegExp(`^[\\.\\${path.sep}]+`), '')
  const normalizedName = normalizedUrl.length === 0 ? filename : `${normalizedUrl.replace(pathSepPattern, '_')}_${filename}`
  const longname = `${normalizedName.toLowerCase().replace(/[^a-z0-9_]/g, '')}.${type}`;
```

I can create a pull request if necessary.

### Haul Configuration (webpack.haul.js)

I added resolve.modules to add a common folder (and full path to node_modules is also required when importing from common).
```
import path from 'path'

module.exports = ({ platform }, defaults) => ({
  ...defaults,
  entry: `./index.js`,
  resolve: {
    ...defaults.resolve,
    modules: [
      path.resolve(__dirname, '../common'),
      path.resolve(__dirname, './node_modules')
    ]
  }
});
```
### Your Environment

| software         | version
| ---------------- | -------
| Haul             |  1.0.0-beta.13
| react-native     |  0.53.0
| node             |  8.9.1
| npm or yarn      | 1.3.2

**Comment by thymikee** [comment]
Anybody interested in taking a look at the provided repro? :)

**Comment by grahamlyus** [comment]
@thymikee If it'd help, I'll try to create a repro repo. I'm pretty slammed this week. I've been using a patch to work around the issue in the meantime.

**Comment by thymikee** [comment]
Closed via #436
